kotlin/manage-operational-layers/src/main/java/com/esri/arcgisruntime/sample/manageoperationallayers/MainActivity.kt

```diff
@@ -0,0 +1,161 @@
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_main)
+
+    // create a map with a topographic basemap
+    val map = ArcGISMap(Basemap.Type.TOPOGRAPHIC, 34.056295, -117.195800, 4)
+
+    val imageLayerElevation =
+      ArcGISMapImageLayer("https://sampleserver5.arcgisonline.com/arcgis/rest/services/Elevation/WorldElevations/MapServer")
+    val imageLayerCensus =
+      ArcGISMapImageLayer("https://sampleserver5.arcgisonline.com/arcgis/rest/services/Census/MapServer")
+    val imageLayerDamage =
+      ArcGISMapImageLayer("https://sampleserver5.arcgisonline.com/arcgis/rest/services/DamageAssessment/MapServer")
+
+    // add the map image layers to the map's operational layers
+    map.operationalLayers.addAll(listOf(imageLayerElevation, imageLayerCensus, imageLayerDamage))
+
+    // set the map to the map view
+    mapView.map = map
+
+    // handle the floating action button and recycler view logic for this sample
+    setupUI()
+  }
+
+  /**
+   * Changes a layer's ordered position within the map's operational layers.
+   *
+   * @param oldPosition the layer's position before the move
+   * @param targetPosition the layer's position after the move
+   */
+  private fun moveLayerFromToPosition(oldPosition: Int, targetPosition: Int) {
+    val operationalLayers = mapView.map.operationalLayers
+    // remove the layer from the map
+    val layer = operationalLayers.removeAt(oldPosition)
+    // add the layer back to the map at the target position
+    operationalLayers.add(targetPosition, layer)
+
+    // tell the recycler view that the item has moved
+    activeRecyclerView.adapter?.notifyItemMoved(oldPosition, targetPosition)
+  }
+
+  /**
+   * Removes a layer from the map's operational layers and stores it the removed layers list.
+   *
+   * @param position the index of the layer in the map's operational layers LayerList
+   */
+  private fun removeLayerFromMap(position: Int) {
+    val operationalLayers = mapView.map.operationalLayers
+    // store the layer in a list to keep track of which layers have been removed
+    removedLayers.add(operationalLayers[position])
+    // remove the layer from the map
+    operationalLayers.removeAt(position)
+
+    // tell the recycler view that the item has been removed
+    activeRecyclerView.adapter?.notifyDataSetChanged()
+  }
+
+  /**
+   * Adds a layer from the removed layer list to the map's operational layers.
+   *
+   * @param position the index of the layer within the list of removed layers
+   */
+  private fun addLayerToMap(position: Int) {
+    // remove the layer from the removed layers
+    val layer = removedLayers.removeAt(position)
+    // add the layer to the map
+    mapView.map.operationalLayers.add(layer)
+
+    // notify the recycler views of the change
+    removedRecyclerView.adapter?.notifyDataSetChanged()
+    activeRecyclerView.adapter?.notifyDataSetChanged()
```

Comment:
I guess these could be:

    removedRecyclerView.adapter?.notifyItemRemoved(position)
    activeRecyclerView.adapter?.notifyItemInserted(mapView.map.operationalLayers.size)

but maybe makes it more confusing?

Reply:
I think this is clearer for both of the above! I wasn't aware of these methods